__tests__/utils/desktop.js

```diff
@@ -0,0 +1,20 @@
+import {createInstance} from 'osjs';
+import {resourceResolver} from '../../src/utils/desktop';
+
+describe('#resourceResolver icon themes', () => {
+  let core;
+
+  beforeAll(() => createInstance().then((c) => (core = c)));
+  afterAll(() => core.destroy());
+
+  test('Should work with image extensions in the icon name', () => {
```

Comment:
This should also test for no extensions, filenames with dots in them, and double extensions.

Reply:
> double extensions

To confirm your desired behavior, would you want `icon-name.gif.png` to become `icon-name` or `icon-name.gif`?

Reply:
> To confirm your desired behavior

The *required* behavior :)


Since this introduces extension resolution, all input has to be stripped of extensions. Preferably a set of known ones (as mentioned in an earlier discussion). The easiest way is probably just to use a regex that removes any supported extensions and the preceding dot.

So:

* `icon-name.png` -> `icon-name`
* `icon-name.foo` -> `icon-name.foo`
* `icon-name.png.png` -> `icon-name.png` (if someone does this, then it's on them imo. super edge-case)
* `icon-name.foo.png` -< `icon-name.foo`
* and so on

Reply:
Alright that's what I already wrote the tests for, perfect. Will make a commit soon 👍🏼